tripal_chado/includes/TripalFields/sbo__relationship/sbo__relationship.inc

```diff
@@ -183,155 +205,205 @@ class sbo__relationship extends ChadoField {
     }
 
-    // Add in the TripalEntity ids if these base records in the relationship
-    // are published.
-    if (property_exists($relationship->$subject_id_key, 'entity_id')) {
-      $entity_id = $relationship->$subject_id_key->entity_id;
-      $entity->{$field_name}['und'][$delta]['value']['local:relationship_subject']['entity'] = 'TripalEntity:' . $entity_id;
+    // Add in the TripalEntity ids if the object is published.
+    if (property_exists($relationship->{$this->subject_id_column}, 'entity_id')) {
+      $entity_id = $relationship->{$this->subject_id_column}->entity_id;
+      $record['entity'] = 'TripalEntity:' . $entity_id;
     }
-    if (property_exists($relationship->$object_id_key, 'entity_id')) {
-      $entity_id = $relationship->$object_id_key->entity_id;
-      $entity->{$field_name}['und'][$delta]['value']['local:relationship_object']['entity'] = 'TripalEntity:' . $entity_id;
+
+    return $record;
+  }
+
+  /**
+   * Retrive the object from the current relationship.
+   */
+  private function getRelationshipObject($relationship) {
+    $name = [];
+
+    // Retrieve the name (may be multiple parts).
+    foreach ($this->base_name_columns as $column) {
+      $name[] = $relationship->{$this->object_id_column}->{$column};
     }
 
+    // Retrieve the Type.
+    $type = $this->instance['settings']['base_table'];
+    if ((!$this->base_type_column == 'table_name') AND isset($relationship->{$this->object_id_column}->{$this->base_type_column})) {
+      $type_object = $relationship->{$this->object_id_column}->{$this->base_type_column};
+      if (isset($type_object->name)) {
+        $type = $type_object->name;
+      }
+      elseif (isset($type_object->uniquename)) {
+        $type = $type_object->uniquename;
+      }
+    }
+
+    $record = [
+      'rdfs:type' => $type,
+      'schema:name' => implode(' ', $name),
+      // @todo support the entity and determine whether this is current one or not.
+      //'entity' => 'TripalEntity:' . $entity->id,
+    ];
+
+    // If the object has a unqiuename then add that in for refernce.
+    if (property_exists($relationship->{$this->object_id_column}, 'uniquename')) {
+      $record['data:0842'] = $relationship->{$this->object_id_column}->uniquename;
+    }
+
+    // If the object has an organism then add that in for reference.
+    if (property_exists($relationship->{$this->object_id_column}, 'organism_id')
+      AND is_object($relationship->{$this->object_id_column}->organism_id)) {
+        $record['OBI:0100026'] = $relationship->{$this->object_id_column}->organism_id->genus . ' ' . $relationship->{$this->object_id_column}->organism_id->species;
+    }
+
+    // Add in the TripalEntity ids if the object is published.
+    if (property_exists($relationship->{$this->object_id_column}, 'entity_id')) {
+      $entity_id = $relationship->{$this->object_id_column}->entity_id;
+      $record['entity'] = 'TripalEntity:' . $entity_id;
+    }
+
+    return $record;
+  }
+
```

Comment:
missing param docs... i feel like entity/delta are pretty self explanatory but not relationship

Reply:
Documentation added :-) 👍